src/scripts/main.js

```diff
@@ -1,3 +1,270 @@
+  resultSortOffice.forEach((row) => tbody.appendChild(row));
+}
+
+function salary() {
+  const sortElements = Array.from(document.querySelectorAll('tbody tr'));
+  const resultSortSalary = sortElements.sort((rowA, rowB) => {
+    const cellA = toValidNum(rowA.cells[4].textContent);
+    const cellB = toValidNum(rowB.cells[4].textContent);
+
+    return callState.countSalary === 0 ? cellA - cellB : cellB - cellA;
+  });
+
+  tbody.innerHTML = '';
+  resultSortSalary.forEach((row) => tbody.appendChild(row));
+}
+
+const form = document.createElement('form');
+
+document.body.append(form);
+form.classList.add('new-employee-form');
+
+form.innerHTML = `<label>Name: <input data-qa="name" type="text"></label>
+<label>Position: <input data-qa="position" type="text"></label>
+<label>Office:
+  <select>
+    <option value="Tokyo">Tokyo</option>
+    <option value="Singapore">Singapore</option>
+    <option value="London">London</option>
+    <option value="NewYork">New York</option>
+    <option value="Edinburgh">Edinburgh</option>
+    <option value="SanFrancisco">San Francisco</option>
+  </select>
+</label>
+<label>Age: <input data-qa="age" type="number"></label>
+<label>Salary: <input data-qa="salary" type="number"></label>
+<button>Save to table</button>`;
+
+form.addEventListener('submit', saveToTable);
+
+function saveToTable(events) {
+  events.preventDefault();
+
+  const nameInput = form.querySelector('input[data-qa="name"]');
+  const positionInput = form.querySelector('input[data-qa="position"]');
+  const ageInput = form.querySelector('input[data-qa="age"]');
+  const salaryInput = form.querySelector('input[data-qa="salary"]');
+  const office = form.querySelector('select');
+  let isValid = true;
+
+  const resultMessage = document.createElement('div');
+
+  document.body.append(resultMessage);
+  resultMessage.setAttribute('data-qa', 'notification');
+  resultMessage.classList.add('notification');
+  resultMessage.style.cssText = 'transition: 1s';
+
+  setTimeout(() => {
+    resultMessage.remove();
+  }, 2000);
+
+  if (nameInput.value.length < 4) {
+    resultMessage.innerHTML = '<p>Name value has less than 4 letters</p>';
+    resultMessage.classList.add('error');
+    isValid = false;
+
+    return;
+  }
+
+  if (positionInput.value.trim() === '') {
+    resultMessage.innerHTML = '<p>Position is required</p>';
+    resultMessage.classList.add('error');
+    isValid = false;
+
+    return;
+  }
+
+  if (ageInput.value < 18 || ageInput.value > 90) {
+    resultMessage.innerHTML = '<p>Age must be a valid</p>';
+    resultMessage.classList.add('error');
+    isValid = false;
+
+    return;
+  }
+
+  if (salaryInput.value === '') {
+    resultMessage.innerHTML = '<p>Add salary</p>';
+    resultMessage.classList.add('error');
+    isValid = false;
+
+    return;
+  }
+
+  function createValidSalary(salaryValue) {
+    const value = salaryValue.toString();
+
+    return `$${value.slice(0, value.length - 3)},${value.slice(value.length - 3)}`;
```

Comment:
The `createValidSalary` function may not correctly format salaries with more than three digits before the comma. Consider using a more robust method for formatting numbers as currency, such as `Intl.NumberFormat`.